flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroRowDataDeserializationSchema.java

```diff
@@ -0,0 +1,371 @@
+				return avroObject -> StringData.fromString(avroObject.toString());
+			case BINARY:
+			case VARBINARY:
+				return this::convertToBytes;
+			case DECIMAL:
+				return createDecimalConverter((DecimalType) type);
+			case ARRAY:
+				return createArrayConverter((ArrayType) type);
+			case ROW:
+				return createRowConverter((RowType) type);
+			case MAP:
+			case MULTISET:
+				return createMapConverter((MapType) type);
+			case RAW:
+			default:
+				throw new UnsupportedOperationException("Unsupported type: " + type);
+		}
+	}
+
+	private DeserializationRuntimeConverter createDecimalConverter(DecimalType decimalType) {
+		return avroObject -> {
+			final byte[] bytes;
+			if (avroObject instanceof GenericFixed) {
+				bytes = ((GenericFixed) avroObject).bytes();
+			} else if (avroObject instanceof ByteBuffer) {
+				ByteBuffer byteBuffer = (ByteBuffer) avroObject;
+				bytes = new byte[byteBuffer.remaining()];
+				byteBuffer.get(bytes);
+			} else {
+				bytes = (byte[]) avroObject;
+			}
+			return DecimalData.fromUnscaledBytes(bytes, decimalType.getPrecision(), decimalType.getScale());
+		};
+	}
+
+	private DeserializationRuntimeConverter createArrayConverter(ArrayType arrayType) {
+		final DeserializationRuntimeConverter elementConverter = createNullableConverter(arrayType.getElementType());
+		final Class<?> elementClass = LogicalTypeUtils.toInternalConversionClass(arrayType.getElementType());
+
+		return avroObject -> {
+			final List<?> list = (List<?>) avroObject;
+			final int length = list.size();
+			final Object[] array = (Object[]) Array.newInstance(elementClass, length);
+			for (int i = 0; i < length; ++i) {
+				array[i] = elementConverter.convert(list.get(i));
+			}
+			return new GenericArrayData(array);
+		};
+	}
+
+	private DeserializationRuntimeConverter createMapConverter(MapType mapType) {
+		LogicalType keyType = mapType.getKeyType();
+		if (!LogicalTypeChecks.hasFamily(keyType, LogicalTypeFamily.CHARACTER_STRING)) {
+			throw new UnsupportedOperationException(
+				"Avro format doesn't support non-string as key type of map. " +
+					"The map type is: " + mapType.asSummaryString());
+		}
+		final DeserializationRuntimeConverter keyConverter = createConverter(keyType);
+		final DeserializationRuntimeConverter valueConverter = createConverter(mapType.getValueType());
+
+		return avroObject -> {
+			final Map<?, ?> map = (Map<?, ?>) avroObject;
+			Map<Object, Object> result = new HashMap<>();
+			for (Map.Entry<?, ?> entry : map.entrySet()) {
+				Object key = keyConverter.convert(entry.getKey());
+				Object value = valueConverter.convert(entry.getValue());
+				result.put(key, value);
+			}
+			return new GenericMapData(result);
+		};
+	}
+
+	private TimestampData convertToTimestamp(Object object) {
+		final long millis;
+		if (object instanceof Long) {
+			millis = (Long) object;
+		} else {
+			// use 'provided' Joda time
+			final DateTime value = (DateTime) object;
+			millis = value.toDate().getTime();
```

Comment:
This might be not correct, because the `java.util.Date` will use local time zone. 
A simpler way to to this is convert it into sql Timestamp first: 

```java
TimestampData.fromTimestamp(new java.sql.Timestamp(value.getMillis());
```

Reply:
To be compatible with the old behavior, i think we should return `TimestampData.fromTimestamp(new Timestamp(millis))` instead to outputs the timestamp without timezone.